package.json

```diff
@@ -636,6 +636,20 @@
           "description": "%java.debugger.configuration.forceBuildBeforeLaunch%",
           "default": true
         },
+        "java.debug.settings.onBuildFailure": {
+          "type": "string",
+          "enum": [
+            "Proceed",
+            "Fix...",
+            "Cancel"
```

Comment:
When build fails, pop up `Fix` or `Cancel` silently is not a good behavior. I would suggest the values to be [`askUser`, `proceed`]. Defaults to `askUser` to show current behavior.

Reply:
Thx for the suggestion. I actually used "no value" as the indicator for "asking the user". So my implementation right now defaults to the current behaviour if you don't use this settings. Only if you do set this in your config will it override current behavior with the option from the settings... is that not better than having an explicit "do not override" setting?

Reply:
You're using null to implicitly indicate something, that's not intuitive. Why not to add it to the enum directly? If you follow the VS Code settings, adding all possible values to the enum is best practice.

Also, please remove `Fix...` and `Cancel` from the settings. When i click debugging, pop up a `Fix` dropdown suddenly or cancel my operation silently is weird.

Reply:
I don't agree. I think it's shortsighted to limit the already available options. If someone for some reason explicitly wants to pop the "Fix..." UI automatically I don't think it should be limited just because it might look "weird" at this point. But I'll just go ahead and change this to move this forward.